src/hotspot/share/cds/heapShared.cpp

```diff
@@ -856,6 +991,52 @@ void HeapShared::resolve_classes_for_subgraphs(JavaThread* current, ArchivableSt
   }
 }
 
+// The main purpose of this call is to initialize any classes that are reachable
+// from the archived Java mirrors that belong to the <class_loader>.
+//
+// For example, if this enum class is initialized at AOT cache assembly time:
+//
+//    enum Fruit {
+//       APPLE, ORANGE, BANANA;
+//       static final Set<Fruit> HAVE_SEEDS = new HashSet<>(Arrays.asList(APPLE, ORANGE));
+//   }
+//
+// the pre-inited mirror of Fruit references HashSet, which should be initialized
+// before any Java code can access the Fruit class. Note that HashSet itself doesn't
+// necessary need to be an aot-initialized class.
+//
+// The set of classes that are required to be initialized for the archived
+// java mirrors are recorded in _runtime_default_subgraph_info (which probably
```

Comment:
_runtime_default_subgraph_info is derived from _default_subgraph_info, the latter being the subgraph associated with class java.lang.Object. Perhaps using the prefix `root` or `jlobject` would make this clearer i.e. `_root_subgraph_info` and `_runtime_root_subgraph_info` or `_jlobject_subgraph_info` and `_runtime_jlobject_subgraph_info`.

Reply:
`root` may be a little over-loaded, because we have `HeapShared::get_root(int)` to get an arbitrary object with an integer index.

I renamed the various field/functions to `special_subgraph`, and added a comment about what this subgraph contains. What do you think?

Also, the two `subgraph_info` fields are renamed to the following:

```
  static KlassSubGraphInfo* _dumptime_special_subgraph;              // for collecting info during dump time
  static ArchivedKlassSubGraphInfoRecord* _runtime_special_subgraph; // for initializing classes during run time.
```

Reply:
I agree `root` is too ambiguous. I think renaming the fields and methods to use `special_subgraph` works well and the comments make it very clear how the dump time and runtime subgraphs are used.